include/circt/Dialect/Moore/MooreOps.td

```diff
@@ -1006,12 +1008,21 @@ def StructExtractOp : MooreOp<"struct_extract"> {
     ```
     See IEEE 1800-2017 § 7.2.1 "Assigning to structures".
   }];
-  let arguments = (ins StrAttr:$memberName, UnpackedType:$input);
+  let arguments = (ins StrAttr:$fieldName, UnpackedType:$input);
   let results = (outs UnpackedType:$result);
   let assemblyFormat = [{
-  $input `,`   $memberName  attr-dict `:`
-  type($input) `->`
-  type($result)
+  $input `,`   $fieldName  attr-dict `:`
+  type($input) `->` type($result)
+  }];
+}
+
+def StructExtractRefOp : MooreOp<"struct_extract_ref"> {
+  let summary = "Struct Extract operation";
+  let arguments = (ins StrAttr:$fieldName, RefType:$input);
+  let results = (outs RefType:$result);
```

Comment:
Can you make `$input` an `AnyStructRefType`, and can you add a constraint like `TypesMatchWith<...>` that ensures the type of `$result` is the type of the accessed field in `$input` wrapped in a `ref<...>`? This will then allow you to remove the `-> type($result)` in the assembly format, since the constraint allows MLIR to determine the type automatically.

Reply:
Thanks for the advice. I have just add a verifier, please check.